File: py/1215.py
def is_alpha(c):
    i = ord(c)

    if (65 <= i <= 90) or (97 <= i <= 122):
        return True

    return False


def andy():
    dic = set()

    while True:
        try:
            line = input()
            word = ''

            for i in line:
                if is_alpha(i):
                    word += i.lower()

                elif len(word) > 0:
                    dic.add(word)
                    word = ''

                else:
                    word = ''

            if len(word) > 0:
                dic.add(word)
                word = ''

        except EOFError:
            print(*sorted(dic), sep='\n')
            break


andy()

File: cpp/1215.cpp
#include <iostream>
#include <string>
#include <set>

using namespace std;

bool is_alpha(char c)
{
    if (('a' <= c && c <= 'z') or ('A' <= c && c <= 'Z')) {
        return true;
    }

    return false;
}

int main()
{
    int i;
    string line, word = "";
    set<string> dict;

    while (getline(cin, line)) {
        for (i = 0; i < line.length(); i++) {
            if (is_alpha(line[i])) {
                word.push_back(tolower(line[i]));
            } else if (word.length() > 0) {
                dict.insert(word);
                word = "";
            } else {
                word = "";
            }
        }

        if (word.length() > 0) {
            dict.insert(word);
            word = "";
        }
    }

    for (auto k: dict) {
        cout << k << "\n";
    }

    return 0;
}